### PR TITLE
Fix sulu 2.0 compatibility, update dependencies and improve ci checks

### LINKED ISSUE #118: validation.xml not up to date with Sulu 2.0
| Q | A
| --- | ---
| Bug? | yes
| New Feature? | no
| Community Bundle Version | 17095b5
| Sulu Version | 2.0.0

#### Actual Behavior

Error during cache:clear on composer auto-script :
> In PropertyMetadata.php line 40:
> Property "country" does not exist in class "Sulu\Bundle\ContactBundle\Entity\Address"

#### Possible Solutions

Solving by commenting L.96 to L.105 of "community-bundle\Resources\config\validation.xml"

```
<class name="Sulu\Bundle\ContactBundle\Entity\Address">
    <property name="country">
        <constraint name="NotBlank">
            <option name="groups">
                <value>registration</value>
                <value>profile</value>
            </option>
        </constraint>
    </property>
</class>
```

### PR DESCRIPTION
**Pull request by alexander-schranz**
| Q | A
| --- | ---
| Bug fix? | no
| New feature? | no
| BC breaks? | no
| Deprecations? | no
| Fixed tickets | fixes #118
| Related issues/PRs | -
| License | MIT

#### What's in this PR?

The country field was removed from sulu 2.0 and need to migrate.

#### Why?

Fixing compatibility.

### CODE REVIEW

**Comment by niklasnatter** [comment]
you need to update the jackalope version to fix the tests 🙂 
https://github.com/sulu/SuluCommentBundle/pull/34/commits/9b1102d82eb823922c38d08f00e4b29fb13cf73f

File: phpstan.neon [hunk 1]
```diff
@@ -5,7 +5,39 @@ includes:
     - vendor/phpstan/phpstan-phpunit/extension.neon
     - vendor/phpstan/phpstan-phpunit/rules.neon
 
+services:
```
Comment (niklasnatter) [inline phpstan.neon]: is there a rule of the `phpstan-strict-rules` package that we dont want to use?

**Comment by nnatter** [comment]
i think we can disable styleci here too as we are using php-cs-fixer now?

File: Tests/Unit/Controller/SaveMediaTraitTest.php [hunk 2]
```diff
@@ -29,70 +29,21 @@ class SaveMediaTraitTest extends TestCase
         getSystemCollectionManager as mockedGetSystemCollectionManager;
     }
 
-    /**
-     * @var MediaManagerInterface
-     */
     private $mediaManager;
```
Comment (niklasnatter) [inline Tests/Unit/Controller/SaveMediaTraitTest.php:36]: i think we should use `ObjectProphecy|MediaManagerInterface` as type for these properties 🙂

File: phpstan.neon (hunk 1 above)
Comment (alexander-schranz) [reply phpstan.neon]: Fixed

**Comment by niklasnatter** [comment]
ping

File: Tests/Unit/Controller/SaveMediaTraitTest.php (hunk 2 above)
Comment (alexander-schranz) [reply Tests/Unit/Controller/SaveMediaTraitTest.php:36]: done

**Pull request merged**